spec/erb_lint/linters/require_script_nonce_spec.rb

```diff
@@ -0,0 +1,167 @@
+        let(:mime_type) { 'type="text/whatever"' }
+
+        it "passes the nonce check" do
+          expect(subject).to(eq([]))
+        end
+      end
+    end
+
+    context 'when nonce has no value' do
+      let(:nonce) { 'nonce' }
+
+      context 'when MIME type is text/javascript' do
+        let(:mime_type) { 'type="text/javascript"' }
+
+        it "builds an offense for a HTML script tag with a missing nonce" do
+          expect(subject).to(eq([
+            build_offense(1..6, html_nonce_message),
+          ]))
+        end
+      end
+
+      context 'when MIME type is not specified' do
+        let(:mime_type) { nil }
+
+        it "builds an offense for a HTML script tag with a missing nonce" do
+          expect(subject).to(eq([
+            build_offense(1..6, html_nonce_message),
+          ]))
+        end
+      end
+
+      context 'when MIME type is not text/javascript' do
+        let(:mime_type) { 'type="text/whatever"' }
+
+        it "passes the nonce check" do
+          expect(subject).to(eq([]))
+        end
+      end
+    end
+
+    context 'when nonce is not present' do
+      let(:nonce) { nil }
+
+      it "builds an offense for a HTML script tag with a missing nonce" do
+        expect(subject).to(eq([
+          build_offense(1..6, html_nonce_message),
+        ]))
+      end
+    end
+  end
+
+  describe 'Javascript helper tags linting' do
+    context 'usage of javascript_tag helper without nonce' do
+      let(:file) { <<~FILE }
+        <br />
+        <%= javascript_tag do %>
+        FILE
+
+      it "builds an offense for a Rails helper script tag with a missing nonce" do
+        expect(subject).to(eq([build_offense(7..30, tag_helper_nonce_message)]))
+      end
+    end
+
+    context 'usage of javascript_include_tag helper without nonce' do
+      let(:file) { <<~FILE }
+        <br />
+        <%= javascript_include_tag "script" %>
+        FILE
+
+      it "builds an offense for a Rails helper script tag with a missing nonce" do
+        expect(subject).to(eq([build_offense(7..44, tag_helper_nonce_message)]))
+      end
+    end
+
+    context 'usage of javascript_pack_tag helper without nonce' do
+      let(:file) { <<~FILE }
+        <br />
+        <%= javascript_pack_tag "script" %>
+        FILE
+
+      it "builds an offense for a Rails helper script tag with a missing nonce" do
+        expect(subject).to(eq([build_offense(7..41, tag_helper_nonce_message)]))
+      end
+    end
+
+    context 'usage of javascript_tag helper with a well formed nonce tag' do
+      let(:file) { <<~FILE }
+        <br />
+        <%= javascript_tag do nonce: true %>
+        FILE
+
+      it "passes the nonce check" do
+        expect(subject).to(eq([]))
+      end
+    end
+
+    context 'usage of javascript_include_tag helper with a well formed nonce tag' do
+      let(:file) { <<~FILE }
+        <br />
+        <%= javascript_include_tag "script" nonce: true %>
```

Comment:
There is a syntax error here. I was expecting this to fail
```suggestion
        <%= javascript_include_tag "script", nonce: true %>
```

Reply:
Good catch! Not sure why the parser didn't choke on this one, but it is updated now :)